tests/src/Kernel/CollaboraMediaAccessTest.php

```diff
@@ -0,0 +1,371 @@
+                'preview own unpublished document in collabora',
+            ]),
+            'Editor' => $this->createUser([
+                'edit any document in collabora',
+            ]),
+            'Kelly (edit own)' => $this->createUser([
+                'edit own document in collabora',
+            ]),
+            'Media admin' => $this->createUser([
+                'administer media',
+            ]),
+        ];
+
+        /** @var \Drupal\media\MediaInterface[] $media_entities */
+        $media_entities = [
+            "Sean's published document" => $this->createMediaEntity('document', [
+                'uid' => $accounts['Sean (preview own)']->id(),
+            ]),
+            "Sean's unpublished document" => $this->createMediaEntity('document', [
+                'uid' => $accounts['Sean (preview own)']->id(),
+                'status' => 0,
+            ]),
+            "Kelly's published document" => $this->createMediaEntity('document', [
+                'uid' => $accounts['Kelly (edit own)']->id(),
+            ]),
+            "Kelly's unpublished document" => $this->createMediaEntity('document', [
+                'uid' => $accounts['Kelly (edit own)']->id(),
+                'status' => 0,
+            ]),
+        ];
+
+        // The users created above are denied access for all the operations.
+        $this->assertEntityAccess(
+            [
+                'anonymous' => [],
+                'authenticated' => [],
+                'media permissions only' => [],
+                'Bookworm' => [],
+                'Previewer' => [
+                    "Sean's published document" => ['preview in collabora'],
+                    "Kelly's published document" => ['preview in collabora'],
+                ],
+                'Sean (preview own)' => [
+                    "Sean's unpublished document" => ['preview in collabora'],
+                ],
+                'Editor' => [
+                    "Sean's published document" => ['edit in collabora'],
+                    "Sean's unpublished document" => ['edit in collabora'],
+                    "Kelly's published document" => ['edit in collabora'],
+                    "Kelly's unpublished document" => ['edit in collabora'],
+                ],
+                'Kelly (edit own)' => [
+                    "Kelly's published document" => ['edit in collabora'],
+                    "Kelly's unpublished document" => ['edit in collabora'],
+                ],
+                'Media admin' => [
+                    "Sean's published document" => ['preview in collabora', 'edit in collabora'],
+                    "Sean's unpublished document" => ['preview in collabora', 'edit in collabora'],
+                    "Kelly's published document" => ['preview in collabora', 'edit in collabora'],
+                    "Kelly's unpublished document" => ['preview in collabora', 'edit in collabora'],
+                ],
+            ],
```

Comment:
I think this new order of array keys is more intuitive.

The cross product strategy causes some kind of redundancy, but that's ok because it runs fast and is not overly verbose.